Framework/PythonInterface/mantid/plots/__init__.py

```diff
@@ -42,6 +42,33 @@
     from mpl_toolkits.mplot3d.axes3d import Axes3D
 
 
+def plot_decorator(func):
+    def wrapper(self, *args, **kwargs):
+        func_value = func(self, *args, **kwargs)
+        # Saves saving it on array objects
+        if mantid.plots.helperfunctions.validate_args(*args, **kwargs):
+            # Fill out kwargs with the values of args
+            for index, arg in enumerate(args):
+                if index is 0:
+                    kwargs["workspaces"] = arg.name()
+                if index is 1:
```

Comment:
Yes it would be but the args don't necessarily exist, if you understand? as they can have no value. This is then skipped by the for loop and not in the the suggested method. 

```Traceback (most recent call last):
  File "/home/sam/Repos/mantid1/qt/applications/workbench/workbench/plugins/workspacewidget.py", line 89, in _do_plot_spectrum
    plot_from_names(names, errors, overplot)
  File "/home/sam/Repos/mantid1/qt/applications/workbench/workbench/plotting/functions.py", line 128, in plot_from_names
    errors=errors, overplot=overplot, fig=fig)
  File "/home/sam/Repos/mantid1/qt/applications/workbench/workbench/plotting/functions.py", line 171, in plot
    plot_fn(ws, **plot_kwargs)
  File "/home/sam/Builds/mantid1/bin/mantid/plots/__init__.py", line 52, in wrapper
    kwargs["spectrum_nums"] = args[1]
IndexError: tuple index out of range
```

Reply:
I see, I overlooked that.